theme/static/js/hs_file_browser.js

```diff
@@ -0,0 +1,969 @@
+                    $('#fb-files-container li').animate({top: 0, left: 0}, 200);    // Custom revert to handle multiple selection
+                    isDragging = false;
+                },
+                drag: function (event, ui) {
+                    $('.ui-selected').css({
+                        top: ui.position.top,
+                        left: ui.position.left
+                    });
+                },
+            }
+        );
+    }
+
+    // Provides selection by drawing a rectangular area
+    $("#fb-files-container")
+        .selectable({
+            filter: "li", cancel: ".ui-selected",
+            stop: function (event, ui) {
+                $(".selection-menu").hide();
+                updateSelectionMenuContext();
+
+                $("#fb-files-container li").removeClass("ui-last-selected");
+                $("#fb-files-container li.ui-selected").first().addClass("ui-last-selected");
+            },
+        });
+
+    // Dismiss right click menu when mouse down outside of it
+    $("#fb-files-container, #fb-files-container li, #fbContainmentWrapper").mousedown(function () {
+        $(".selection-menu").hide();
+    });
+
+    $("#hs-file-browser li.fb-folder").dblclick(onOpenFolder);
+
+    // Right click menu for file browser
+    $("#fbContainmentWrapper").bind("contextmenu", function (event) {
+        // Avoid the real one
+        event.preventDefault();
+
+        var menu;   // The menu to show
+        updateSelectionMenuContext();
+
+        if ($(event.target).closest("li").length) {     // If a file item was clicked
+            if (!$(event.target).closest("li").hasClass("ui-selected")) {
+                $(".ui-selected").removeClass("ui-selected");
+                $(event.target).closest("li").addClass("ui-selected");
+                $("#fb-files-container li").removeClass("ui-last-selected");
+                $(event.target).closest("li").addClass("ui-last-selected");
+            }
+            menu = $("#right-click-menu");
+        }
+        else {
+            menu = $("#right-click-container-menu");    // empty space was clicked
+        }
+
+        $(".selection-menu").hide();    // Hide other menus
+
+        var top = event.pageY;
+        var left = event.pageX;
+
+        menu.css({top: top, left: left});
+
+        if (menu.css("display") == "none") {
+            menu.show();
+        }
+        else {
+            menu.hide();
+        }
+    });
+}
+
+function setBreadCrumbs(path) {
+    var crumbs = $("#fb-bread-crumbs");
+    crumbs.empty();
+
+    if (path.lastIndexOf("/") == "-1") {
+        $("#fb-move-up").toggleClass("disabled", true)
+    }
+    else {
+        $("#fb-move-up").toggleClass("disabled", false)
+    }
+
+    var setFirstActive = false;
+    while (path){
+        var folder = path.substr(path.lastIndexOf("/") + 1, path.length);
+        var currentPath = path;
+        path = path.substr(0, path.lastIndexOf("/"));
```

Comment:
@hyi, the 'path' variable is being used to extract the value for the 'folder' variable in each iteration.
Although it is not expected to be empty, I added a base case conditional in case an error causes it to be empty.